src/view/styles.js

```diff
@@ -0,0 +1,490 @@
+			enumerable: false,
+			writable: true
+		} );
+	}
+
+	/**
+	 * Number of styles defined.
+	 *
+	 * @type {Number}
+	 */
+	get size() {
+		return this.getStyleNames().length;
+	}
+
+	/**
+	 * Re-sets internal styles definition.
+	 *
+	 * @param {String} styleString
+	 */
+	setStyle( styleString ) {
+		this.clear();
+
+		const map = parseInlineStyles( styleString );
+
+		for ( const key of map.keys() ) {
+			const value = map.get( key );
+
+			this.converter.toNormalizedForm( key, value, this._styles );
+		}
+	}
+
+	/**
+	 * Checks if single style rule is set.
+	 *
+	 * Supports shorthands.
+	 *
+	 * @param {String} propertyName
+	 * @returns {Boolean}
+	 */
+	hasProperty( propertyName ) {
+		const normalized = this.converter.getNormalized( propertyName, this._styles );
+
+		if ( !normalized ) {
+			// Try return styles set directly - values that are not parsed.
+			return this._styles[ propertyName ] !== undefined;
+		}
+
+		if ( isObject( normalized ) ) {
+			const styles = this.converter.getReducedForm( propertyName, normalized );
+
+			const propertyDescriptor = styles.find( ( [ property ] ) => property === propertyName );
+
+			// Only return a value if it is set;
+			return Array.isArray( propertyDescriptor );
+		} else {
+			return true;
+		}
+	}
+
+	/**
+	 * Inserts single style property.
+	 *
+	 * Can insert one by one
+	 *
+	 *		styles.insertProperty( 'color', 'blue' );
+	 *		styles.insertProperty( 'margin-right', '1em' );
+	 *
+	 * or many styles at once:
+	 *
+	 *		styles.insertProperty( {
+	 *			color: 'blue',
+	 *			'margin-right': '1em'
+	 *		} );
+	 *
+	 * Supports shorthands.
+	 *
+	 * @param {String|Object} nameOrObject
+	 * @param {String|Object} value
+	 * @returns {Boolean}
+	 */
+	insertProperty( nameOrObject, value ) {
+		if ( isObject( nameOrObject ) ) {
+			for ( const key of Object.keys( nameOrObject ) ) {
+				this.insertProperty( key, nameOrObject[ key ] );
+			}
+		} else {
+			this.converter.toNormalizedForm( nameOrObject, value, this._styles );
+		}
+	}
+
+	/**
+	 * Removes styles property.
+	 *
+	 * @param name
+	 */
+	removeProperty( name ) {
+		unset( this._styles, toPath( name ) );
+		delete this._styles[ name ];
+	}
+
+	/**
+	 * Return normalized style object;
+	 *
+	 *		const styles = new Styles();
+	 *		styles.setStyle( 'margin:1px 2px 3em;' );
+	 *
+	 *		console.log( styles.getNormalized( 'margin' ) );
+	 *		// will log:
+	 *		// {
+	 *		//     top: '1px',
+	 *		//     right: '2px',
+	 *		//     bottom: '3em',
+	 *		//     left: '2px'
+	 *		// }
+	 *
+	 * @param {String} name
+	 * @returns {Object|undefined}
+	 */
+	getNormalized( name ) {
+		return this.converter.getNormalized( name, this._styles );
+	}
+
+	/**
+	 * Returns a string containing normalized styles string or undefined if no style properties are set.
+	 *
+	 * @returns {String|undefined}
+	 */
+	getInlineStyle() {
+		const entries = this._getStylesEntries();
+
+		// Return undefined for empty styles map.
+		if ( !entries.length ) {
+			return;
```

Comment:
I'd rather expect an empty string.

Reply:
Also, `StylesMap.isEmpty` getter to be used at the very beginning. Methods of this class will be used hundreds of time and all of those normalization methods do quite a lot of things to return... an empty thing (which will be 99% of times). So, I'd make sure the "return empty thing" path is quick, just to feel safe.